src/components/message-display-rules/NewRuleForm.tsx

```diff
@@ -0,0 +1,123 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ***************************************************************************** */
+
+import { nanoid } from 'nanoid';
+import React, { useRef, useState } from 'react';
+import { useMessageDisplayRulesStore } from '../../hooks';
+import { MessageDisplayRule, MessageViewType } from '../../models/EventMessage';
+import AutocompleteInput from '../util/AutocompleteInput';
+import Select from '../util/Select';
+
+type NewRuleFormProps = {
+	sessions: string[];
+	rule?: MessageDisplayRule | null;
+	stopEdit?: () => void;
+};
+
+const viewTypes = Object.values(MessageViewType);
+
+const NewRuleForm = ({ rule, stopEdit, sessions }: NewRuleFormProps) => {
+	const rulesStore = useMessageDisplayRulesStore();
+
+	const sessionInputRef = useRef(null);
+	const [currentSessionValue, setSessionValue] = useState(rule ? rule.session : '');
+	const [currentSelectedViewType, setCurrentSelectedViewType] = useState(
+		rule ? rule.viewType : MessageViewType.JSON,
+	);
+
+	const submitHandler = (e: React.MouseEvent) => {
+		e.stopPropagation();
+		if (stopEdit) {
+			stopEdit();
+		}
+		if (!rule) {
+			const newRule = {
+				id: nanoid(),
+				session: currentSessionValue,
+				viewType: currentSelectedViewType,
+				removable: true,
+				fullyEditable: true,
+			};
+			rulesStore.setNewMessagesDisplayRule(newRule);
+			setSessionValue('');
+			return;
+		}
+		const isSame =
+			rule && rule.session === currentSessionValue && rule.viewType === currentSelectedViewType;
+		if (rule.session === '*') {
+			rulesStore.setRootDisplayRule({
+				...rule,
+				viewType: currentSelectedViewType,
+			});
+			return;
+		}
+		if (!isSame) {
+			const newRule = {
+				id: rule.id,
+				session: currentSessionValue,
+				viewType: currentSelectedViewType,
+				removable: rule.removable,
+				fullyEditable: rule.fullyEditable,
+			};
+			rulesStore.editMessageDisplayRule(rule, newRule);
+		}
+	};
+
+	return (
+		<div className='rule-inputs'>
+			<div className='inputs-wrapper'>
+				<div className='session'>
+					{!rule || rule.fullyEditable ? (
+						<AutocompleteInput
+							autoresize={false}
+							placeholder='New session'
+							className='session-input'
+							ref={sessionInputRef}
+							value={currentSessionValue}
+							onSubmit={v => {
+								if (v === '*') {
+									setSessionValue('');
+									return;
+								}
+								setSessionValue(v);
+							}}
+							notResetOnSubmit
+							autocomplete={sessions}
+							datalistKey='session-input'
+						/>
+					) : (
+						<p>{rule.session}</p>
+					)}
+				</div>
+				<div className='view-type'>
+					<Select
+						options={viewTypes}
+						selected={currentSelectedViewType}
+						onChange={(v: MessageViewType) => setCurrentSelectedViewType(v)}
+					/>
+				</div>
+			</div>
+			<button
+				className='rule-button'
+				disabled={!currentSessionValue}
```

Comment:
Right now autocomplete input updates value only on submit. So if i type something then i cant click submit button because its disabled
i need to click multiple times outside of input so it can become active, we need to do something about it

Reply:
Lets just delete that disabled prop, and inside submit handler we will check if value isnt empty